cmd/mdatagen/internal/testdata/config_gen/optional_string/config.go

```diff
@@ -0,0 +1,30 @@
+// Code generated by github.com/atombender/go-jsonschema, DO NOT EDIT.
+
+package test_pkg
+
+import "encoding/json"
+import "go.opentelemetry.io/collector/config/configoptional"
+
+// Configuration parameters for the batch processor.
+type Config struct {
+	// ExampleOptional corresponds to the JSON schema field "ExampleOptional".
+	ExampleOptional configoptional.Optional[string] `mapstructure:"ExampleOptional"`
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (j *Config) UnmarshalJSON(value []byte) error {
+	var raw map[string]interface{}
+	if err := json.Unmarshal(value, &raw); err != nil {
+		return err
+	}
+	type Plain Config
+	var plain Plain
+	if err := json.Unmarshal(value, &plain); err != nil {
+		return err
+	}
+	if v, ok := raw["ExampleOptional"]; !ok || v == nil {
+		plain.ExampleOptional = "ex_default"
```

Comment:
This line doesn't compile, because unlike `configopaque.String`, with `configoptional.Optional` it's not possible to use the assignment operator like this.